labs/ipfs-naming/api-cli/names-operations.mdx

```diff
@@ -0,0 +1,351 @@
+---
+meta:
+title: Name operations with Scaleway API/CLI
+description: This page explains how to use the Scaleway IPFS Naming API/CLI to manage names.
+content:
+h1: Name operations with Scaleway API/CLI
+paragraph: This page explains how to use the Scaleway IPFS Naming API/CLI to manage names.
+tags: ipfs ipfs-naming naming name labs web3
+dates:
+validation: 2023-11-07
+posted: 2023-11-07
+categories:
+  - labs
+  - naming
+  - ipfs-naming
+---
+
+Created by [Protocol Labs](https://protocol.ai/), InterPlanetary File System (IPFS) is a decentralized protocol used to store and share content.
+
+Scaleway IPFS Naming allows you to create a key to point to your latest update, and above all maintain this information on the IPFS network and share it through Distributed Hash Tables (DHTs), giving you an extra layer of performance in the region of your choice. This way your records remain available and accessible, even when your local machine is offline.
+
+In this quickstart, we will show you how to create and import an IPNS key and update your records using Scaleway IPFS Naming.
+
+## Scaleway Command Line Interface overview
+
+The [Scaleway Command Line Interface (CLI)](https://github.com/scaleway/scaleway-cli) allows you to create and interact with your names. You will be able to create names, manage them, add/delete value within them.
+
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+    - You have an account and are logged into the [Scaleway console](https://console.scaleway.com/)
+    - You have generated your [API key](/identity-and-access-management/iam/how-to/create-api-keys/)
+    - You have downloaded and installed the [Scaleway CLI v2](https://github.com/scaleway/scaleway-cli#scaleway-cli-v2)
+</Message>
+
+## Configuring the CLI
+
+1. Open a terminal and enter the following command, to configure the Scaleway CLI on your machine:
+
+`scw init`
+
+If you have already configured the CLI, you will see a message like the following:
+
+ ```
+ Current config is located at /path/to/config.yaml
+ access_key: <YOUR ACCESS KEY>
+ secret_key: <YOUR SECRET KEY>
+ default_organization_id: <YOUR ORGANIZATION ID>
+ default_project_id: <YOUR PROJECT ID>
+ default_region: <YOUR DEFAULT REGION, eg fr-par>
+ default_zone: <YOUR DEFAULT ZONE, eg fr-par-1>
+
+ Do you want to override the current config? (Y/n):
+```
+<Message type="note">
+    As the project is available on pl-waw, nl-ams and fr-par, you can set the region to pl-waw, nl-ams or fr-par.
+</Message>
```

Comment:
```suggestion
    ```
    scw init
    ```
    
    If you have already configured the CLI, you will see a message like the following:
    
     ```
     Current config is located at /path/to/config.yaml
     access_key: <YOUR ACCESS KEY>
     secret_key: <YOUR SECRET KEY>
     default_organization_id: <YOUR ORGANIZATION ID>
     default_project_id: <YOUR PROJECT ID>
     default_region: <YOUR DEFAULT REGION, eg fr-par>
     default_zone: <YOUR DEFAULT ZONE, eg fr-par-1>
    
     Do you want to override the current config? (Y/n):
    ```
    <Message type="note">
        As the project is available on pl-waw, nl-ams and fr-par, you can set the region to pl-waw, nl-ams or fr-par.
    </Message>
```